### DISCUSSION TITLE
Possibly premature: How is the log API expected to be used?

### DISCUSSION THREAD

**Discussion by io7m** [open-telemetry/opentelemetry-java#5488] [Q&A]
Hello!

I have a fully manually instrumented application that's happily sending out metrics and traces. I'd like to emit log events independent of metrics or traces for the purposes of monitoring significant events that happen during normal operation of the application.

Will the logger and/or event API work similarly to how the metrics and traces APIs work now? By that, I mean, I've got an `OpenTelemetry` instance and I call `getTracer()` and `getMeter()`, etc, to build spans and create meters. Will the logging API work the same way? I assume I'd call `getLogger()` and publish events to it.

I can't seem to find any code examples on this. I assume that's because they don't actually exist yet. :slightly_smiling_face:

**Comment by io7m** [comment]
I should probably add that one thing I like about the way OpenTelemetry is configured is that I can opt-in to metrics, but opt-out of traces if I don't care about those (by, for example, just not configuring an exporter for traces). I could imagine people running this application and _only_ wanting to capture log events and ignore metrics/traces.

**Comment by jkwatson** [comment]
No, the log appender API is strictly intended for the use-case of building appenders for existing Java ecosystem logging APIs. 

We are likely to have an "event" API at some point in the future, but that should end up having a much more restricted API than a full-fledged logging API, and wouldn't be intended as a logging API solution, either.

We do not intend to invent yet another logging API for Java, but instead provide means to write appenders that can output OTLP log records via the OpenTelemetry SDK.

**Comment by io7m** [comment]
The "event" API is what I was interested in. I assumed it would be introduced along with the logging API given that events are log records underneath (according to the OTLP documentation).

I'm actually not interested in the slightest in replacing something like SLF4J... It's barely used in this project anyway; anything that matters enough to be logged goes out via telemetry instead.

**Reply by jkwatson** [reply]
The event API is still a work-in-progess in the specification, so until that stabilizes, we won't know for sure what the API is going to look like.

**Comment by io7m** [comment]
Thanks!
